pallets/messages/src/lib.rs

```diff
@@ -417,4 +424,43 @@ impl<T: Config> Pallet<T> {
 		BlockMessages::<T>::set(BoundedVec::default());
 		T::WeightInfo::on_initialize(message_count, schema_count)
 	}
+
+	fn map_error_and_weight(error: DispatchError, weight: Weight) -> DispatchErrorWithPostInfo {
+		DispatchErrorWithPostInfo {
+			post_info: PostDispatchInfo {
+				actual_weight: Some(weight),
+				pays_fee: Default::default(),
+			},
+			error,
+		}
+	}
+
+	fn map_on_chain_errors(result: DispatchResultWithPostInfo) -> DispatchResultWithPostInfo {
+		match result {
+			Ok(p) => Ok(p),
+			Err(e) => {
+				if let DispatchError::Module(ModuleError { message: Some(err_msg), .. }) = e.error {
+					if err_msg == Error::<T>::ExceedsMaxMessagePayloadSizeBytes.as_str() {
```

Comment:
I wished I could replace these if elses with match but couldn't find an easy way.

Reply:
These errors are not exhaustive but takes care of most of use cases and we can add more later